common/include/pcl/impl/point_types.hpp

```diff
@@ -1538,19 +1382,23 @@ namespace pcl
   PCL_EXPORTS std::ostream& operator << (std::ostream& os, const ReferenceFrame& p);
   struct EIGEN_ALIGN16 ReferenceFrame : public _ReferenceFrame
   {
-    inline ReferenceFrame (const _ReferenceFrame &p)
+    inline PCL_CONSTEXPR_CTOR ReferenceFrame (const _ReferenceFrame &p) :
+      ReferenceFrame{p.rf}
     {
-      std::copy_n(p.rf, 9, rf);
+      //std::copy_n(p.rf, 9, rf); // this algorithm is constexpr starting from C++20
     }
 
-    inline ReferenceFrame ()
+    inline PCL_CONSTEXPR_CTOR ReferenceFrame () :
+      _ReferenceFrame{ {{0.0f}} }
     {
-      std::fill_n(x_axis, 3, 0.f);
+      // this algorithm is constexpr starting from C++20
+      /*std::fill_n(x_axis, 3, 0.f);
       std::fill_n(y_axis, 3, 0.f);
-      std::fill_n(z_axis, 3, 0.f);
+      std::fill_n(z_axis, 3, 0.f);*/
     }
 
-    // @TODO: add other ctors
+    inline PCL_CONSTEXPR_CTOR ReferenceFrame (const float* _rf) :
```

Comment:
Bad addition. Specially considering:
* No other ctor takes in a pointer
* There is no bounds checking
* The array used to call this is not of sufficient length

Suggestions:
1. Can we use const reference instead of pointer?
2. Can we specify the length of the array in the type? Eg: `char (&p)[10]`.
3. If needed, make a private typename and use it. Eg: `using Ref = float[10]; Ref* p`
4. Can we make this ctor private?

Reply:
1. yes, this actually seems would be the most suitable signature
2. yes
3. don't think it's neccessary, it's only used once
4. -

Reply:
Private ctor is more about not making this enter into the API

Reply:
i've only implemented TODO "add other ctors", according to the way it's done for other types
why interface of this particular class should be different from other point types in this regard?
every other type has 1. constructor taking object of class with name with an underscore at the begging 2. default constructor 3. constructor taking values needed to initialize all class members

Reply:
You're right. If the first 2 of my initial suggestions are added, then no change of visibility are needed